### PR TITLE
Update controller for when made public field.

### LINKED ISSUE #1180: Update when made public timestamp.
**Current behaviour:** Currently we are not updating `when_made_public` timestamp when a user makes his submission public.
**Expected behaviour:** When a user makes his submission public then update this timestamp.
The timestamp must be in` UTC`.

**Dependency:** Dependent on the PR #1179

### PR DESCRIPTION
**Pull request by RishabhJain2018**
fixes #1180. @deshraj Please review the PR.

### CODE REVIEW

File: frontend/src/js/controllers/challengeCtrl.js [hunk 1]
```diff
@@ -888,12 +888,25 @@
         };
 
         vm.changeSubmissionVisibility = function(submission_id) {
+            var dateTime = new Date();
+            dateTime = dateTime.toISOString();
+            var splitDateTime = dateTime.split("T");
+            var date = splitDateTime[0];
+            var time = splitDateTime[1].split(".")[0];
```
Comment (deshraj) [inline frontend/src/js/controllers/challengeCtrl.js]: The datetime should be computed at the backend and not at the frontend. Both are decouped.

Comment (deshraj) [inline frontend/src/js/controllers/challengeCtrl.js]: The datetime should be computed at the backend and not at the frontend. Both frontend and backend are decoupled.

Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeCtrl.js]: okay sure.

File: frontend/src/js/controllers/challengeCtrl.js [hunk 2]
```diff
@@ -888,12 +888,25 @@
...
+
             var parameters = {};
             parameters.url = "jobs/challenge/" + vm.challengeId + "/challenge_phase/" + vm.phaseId + "/submission/" + submission_id;
             parameters.method = 'PATCH';
-            parameters.data = {
-                "is_public": vm.submissionVisibility[submission_id]
-            };
+            if (vm.submissionVisibility[submission_id]) {
+                parameters.data = {
+                    "is_public": vm.submissionVisibility[submission_id],
```
Comment (aayusharora) [inline frontend/src/js/controllers/challengeCtrl.js]: Please store ` vm.submissionVisibility[submission_id]` in  a seperate variable, and use it everywhere @RishabhJain2018 in this function

**Comment by RishabhJain2018** [comment]
Since we will be calculating the time at the backend, so I am closing this one.

**Pull request closed without merging**